practice/earthquake/site.js

```diff
@@ -0,0 +1,169 @@
+ * background on which earthquake tokens will be placed.
+ */
+function createMap() {
+  var map = new ol.Map({
+    view: new ol.View({
+      center: [0, 0],
+      zoom: 1,
+      projection: 'EPSG:4326'
+    }),
+    layers: [
+      new ol.layer.Tile({
+        source: new ol.source.OSM()
+      }),
+    ],
+    target: 'map'
+  });
+  return map;
+}
+
+/**
+ * Calculates the opacity of a token based on its age since time of access.
+ * @param  {number} time Time of earthquake, in UTC time (milliseconds since
+ * 01-01-1970)
+ * @return {number}      Opacity of object based on how recently it ocurred,
+ * with 1 meaning it occurred the moment of access.
+ */
+function calculateOpacity(time) {
+  var weekUTC = 604800000;
+  var timeBeforeAccess = (time - ACCESS_TIME_UTC) * -1;
+  return timeBeforeAccess / weekUTC;
+}
+
+/**
+ * Calculates the scale of a token based on the magnitude size of earthquake.
+ * @param  {number} magnitude Richter scale measurement of the earthquake size
+ *                            (1-10)
+ *         {number} scaleMax  Maximum value of scale for desired visual output.
+ * @return {number}           Scale number of object.
+ */
+function calculateScale(magnitude) {
+  var percentOfMax = magnitude / 10;
+  var scaleMax = .75;
+  return percentOfMax * scaleMax;
+}
+
+/**
+ * Takes in transformed earthquake data and creates a .Feature OpenLayers
+ * object which contains the coordinates of the earthquake, its time, and its
+ * magnitude.
+ * @param  {array} eqCoordsMagsAndTimes Array of arrays containing earthquake data.
+ * @return {Object}                  OpenLayers.Feature object containing data.
+ */
+function getPlotPoints(eqCoordsMagsAndTimes) {
+  return _.map(eqCoordsMagsAndTimes, function(eqCoordMagAndTime) {
+    return new ol.Feature({
+      geometry: new ol.geom.Point([eqCoordMagAndTime[1],eqCoordMagAndTime[2]]),
+      magnitude: eqCoordMagAndTime[0],
+      time: eqCoordMagAndTime[3]
+    });
+  });
+}
+
+/**
+ * Creates a new layer and adds individual points to the layer, applying style
+ * based on the magnitude and time of the earthquake.
+ * @param {array} plotPoints    Array of objects containing each individual
+ * .Feature object which correspond to each earthquake in data.
+ * @param {object} map           Instanced ol.Map object.
+ */
+function addPointsToMap(plotPoints, map) {
+  var vectorSource = new ol.source.Vector({});
+  for (var i = 0; i < plotPoints.length; i += 1) {
+    vectorSource.addFeature(plotPoints[i]);
+  }
+  var newLayer = new ol.layer.Vector({
+    source: vectorSource,
+    style: function(feature) {
+      return new ol.style.Style({
+        image: new ol.style.Icon({
+          opacity: calculateOpacity(feature.get('time')),
+          scale: calculateScale(feature.get('magnitude')),
+          size: [250, 250],
+          src: '../earthquake/earthquake.png'
+        })
+      });
+    }
+  });
+  map.addLayer(newLayer);
+}
```

Comment:
Kind of advanced feedback: all of the verbs surrounding this problem don't actually deal with the `ol.Map`, they are more closely manipulating the `ol.source.Vector`. If the code was structured to set up an `ol.source.Vector` and just pass that around between functions, there would be better isolation.

The general idea is: try to have "need to know basis" code. The more you isolate parts you don't want manipulated, the more robust the code will be when expanded.